eth/vm/logic/arithmetic.py

```diff
@@ -202,11 +200,12 @@ def shr(computation):
     Bitwise right shift
     """
     shift_length, value = computation.stack_pop(num_items=2, type_hint=constants.UINT256)
-    bit_vector = BitVector(intVal=value, size=constants.CONST_256)
 
     if shift_length >= constants.CONST_256:
         result = constants.CONST_0
+    elif shift_length == constants.CONST_0:
+        result = value
     else:
-        result = bit_vector.shift_right(shift_length).int_val()
+        result = (value // 2**shift_length)
```

Comment:
This one errored out with an overflow error saying the integer is too large to convert to the c_type, probably something that could be worked around if we're clever.

Reply:
It works with the same trick `(value >> shift_length) & constants.UINT_256_MAX`

Reply:
viola! huzzah for simple solutions